eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php

```diff
@@ -213,7 +214,12 @@ public function lookup($url)
         } elseif ($urlAliasId === self::NOT_FOUND) {
             throw new NotFoundException('UrlAlias', $url);
         } else {
-            $urlAlias = $this->loadUrlAlias($urlAliasId);
+            try {
+                $urlAlias = $this->loadUrlAlias($urlAliasId);
+            } catch (NotFoundException $e) {
+                $cache->set(self::NOT_FOUND);
+                throw $e;
+            }
```

Comment:
why do you do this?this seems to be something completely different then the issue here, or?
to many things in one PR just makes it impossible to review btw.